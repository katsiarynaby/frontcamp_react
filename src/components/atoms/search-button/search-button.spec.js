import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import SearchButton from './index';

describe('SearchButton', () => {
    it("SearchButton active should render correctly", () => {
        const wrapper = shallow(<SearchButton
            active={true}
            value="SEARCH"
            onClick={jest.fn()}
        />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it("SearchButton doesn't active should render correctly", () => {
        const wrapper = shallow(<SearchButton
            active={false}
            value="SEARCH"
            onClick={jest.fn()}
        />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});


