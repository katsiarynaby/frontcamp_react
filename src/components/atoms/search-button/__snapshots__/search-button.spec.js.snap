// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton SearchButton active should render correctly 1`] = `
<button
  className="search-button--active"
  onClick={[MockFunction]}
  type="button"
  value="SEARCH"
>
  SEARCH
</button>
`;

exports[`SearchButton SearchButton doesn't active should render correctly 1`] = `
<button
  className="search-button"
  onClick={[MockFunction]}
  type="button"
  value="SEARCH"
>
  SEARCH
</button>
`;
